packages/aws-cdk/lib/api/hotswap/monitor/logs-monitor.ts

```diff
@@ -0,0 +1,188 @@
+import * as util from 'util';
+import * as colors from 'colors/safe';
+import { print, error } from '../../../logging';
+import { flatten } from '../../../util/arrays';
+import { ISDK } from '../../aws-auth';
+
+// how often we should read events from CloudWatchLogs
+const SLEEP = 2_000;
+
+/**
+ * Represents a CloudWatch Log Event that will be
+ * printed to the terminal
+ */
+export interface CloudWatchLogEvent {
+  readonly message: string;
+  readonly logGroup: string;
+  readonly timestamp: Date;
+}
+
+/**
+ * Options for creating a CloudWatchLogEventMonitor
+ */
+export interface CloudWatchLogEventMonitorOptions {
+  /**
+   * The time that watch was first triggered.
+   * CloudWatch logs will be filtered using this start time
+   *
+   * @default - the current time
+   */
+  readonly hotswapTime?: Date;
+
+  /**
+   * The EventPrinter to use to print out CloudWatch events
+   *
+   * @default - a default printer will be created
+   */
+  readonly printer?: IEventPrinter;
+}
+
+export class CloudWatchLogEventMonitor {
+  /**
+   * Determines which events not to display
+   */
+  private startTime: number;
+
+  private logGroups = new Set<string>();
+  private sdk?: ISDK;
+
+  private active = true;
```

Comment:
If we're going to have `activate()` / `deactivate()` methods, I feel like the monitor starting in the `active` state is actually kind of surprising.

Maybe `false` is the better default here? (I agree that, if we didn't have `active()` / `deactivate()`, this default would be fine)

Reply:
Done.